tests/src/Search/ElasticTestCase.php

```diff
@@ -0,0 +1,158 @@
+    {
+        $response = $this->getResponse();
+        if (!$response->isOk()) {
+            $this->fail('Response returned was not 200');
+        }
+        $json = json_decode($response->getContent());
+
+        return $json;
+    }
+
+    public function newClient()
+    {
+        return $this->api = static::createClient();
+    }
+
+    public function getResponse()
+    {
+        return $this->api->getResponse();
+    }
+
+    public function addDocumentToElasticSearch(string $doc)
+    {
+        $obj = json_decode($doc);
+        $this->client->indexJsonDocument($obj->type, $obj->id, $doc, true);
+    }
+    public function addDocumentsToElasticSearch(array $docs)
+    {
+        foreach ($docs as $doc) {
+            $this->addDocumentToElasticSearch($doc);
+        }
+    }
+
+    public function createConfiguration()
+    {
+        if (file_exists(__DIR__.'/../../../config/local.php')) {
+            $this->isLocal = true;
+            $config = include __DIR__.'/../../../config/local.php';
+        } else {
+            $this->isLocal = false;
+            $config = include __DIR__.'/../../../config/ci.php';
+        }
+
+        return $config;
+    }
+
+    protected function jsonRequest(string $verb, string $endpoint, array $params = array())
+    {
+        $headers = array(
+            'HTTP_ACCEPT' => 'application/json',
+            'CONTENT_TYPE' => 'application/json',
+        );
+
+        return $this->api->request(
+            $verb,
+            $endpoint,
+            $params,
+            [],
+            $headers
+        );
+    }
+
+    /**
+     * Creates the application.
+     *
+     * @return HttpKernelInterface
+     */
+    public function createApplication()
+    {
+        $this->kernel = new Kernel($this->createConfiguration());
+
+        return $this->kernel->getApp();
+    }
+
+    public function getElasticSearchClient(callable $fn = null): ElasticsearchClient
+    {
+        return $fn ? $fn($this->kernel->get('elastic.client')) : $this->kernel->get('elastic.client');
+    }
+
+    public function setUp()
+    {
+        parent::setUp();
+        $this->client = $this->getElasticSearchClient();
+        $lines = $this->runCommand('search:setup');
+        if ($lines[0] === 'No alive nodes found in your cluster') {
+            $this->markTestSkipped('Elasticsearch may not be installed, skipping');
+        }
+        $this->assertStringStartsWith('Created new index', $lines[0], 'Failed to run test during set up');
+    }
+
+    public function tearDown()
+    {
+        $this->client->deleteIndex();
+        parent::tearDown();
+    }
+
+    public function runCommand(string $command)
+    {
+        $log = $this->returnCallback(function ($message) use (&$logs) {
+            $logs[] = $message;
+        });
+        $logs = [];
+        $logger = $this->createMock(NullLogger::class);
+
+        foreach (['debug', 'info', 'alert', 'notice', 'error'] as $level) {
```

Comment:
missing `warning` and `critical` that we are using, plus `emergency` and similar, go for the full list